### PR TITLE
Enhancements around convention tables

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: tempto-core/src/main/java/io/prestodb/tempto/internal/convention/ProcessUtils.java [hunk 1]
```diff
@@ -14,19 +14,26 @@
...
+import java.util.function.Consumer;
 
 import static com.google.common.base.Preconditions.checkState;
 
 public final class ProcessUtils
 {
     private static final int SUCCESS_EXIT_CODE = 0;
 
-    public static void execute(String... cmdarray)
+    public static void execute(String ... cmdarray)
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestodb/tempto/internal/convention/ProcessUtils.java]: unnecessary change

File: tempto-core/src/main/java/io/prestodb/tempto/internal/convention/ProcessUtils.java [hunk 2]
```diff
@@ -14,19 +14,26 @@
...
+    {
+        execute(process -> {}, cmdarray);
+    }
+
+    public static void execute(Consumer<Process> processConsumer, String ... cmdarray)
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestodb/tempto/internal/convention/ProcessUtils.java]: fmt

File: tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java [hunk 3]
```diff
@@ -95,6 +106,16 @@ public ConventionTableDefinitionDescriptor(Path ddlFile)
         }
     }
 
+    private long copy(Path dataFile, Process process)
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java]: `s/copy/feed/`

File: tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java [hunk 4]
```diff
@@ -83,6 +87,13 @@ public ConventionTableDefinitionDescriptor(Path ddlFile)
         this.ddlFile = ddlFile;
 
         Path dataFile = changeExtension(ddlFile, "data");
+        Path dataGeneratorFile = changeExtension(ddlFile, "data-generator");
+        if (exists(dataGeneratorFile)) {
+            checkArgument(!exists(dataFile), "Expected no data file when data-generator file is present");
+            makeExecutable(dataGeneratorFile);
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java]: ouch. This should be read-only.

Comment (kokosing) [reply tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java]: It is a temporary copy of read only files

File: tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java (hunk 3 above)
Comment (kokosing) [reply tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java]: hm... it is just a wrapper of `Files.copy`. It just suppresses the exception.

Comment (findepi) [reply tempto-core/src/main/java/io/prestodb/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionDescriptor.java]: But is very specific usage: to feed input to a process. You rather don't call it "copying".
